Sources/MapboxMaps/Gestures/GestureHandlers/PanGestureHandler.swift

```diff
@@ -33,38 +39,63 @@ internal final class PanGestureHandler: GestureHandler {
             cameraAnimationsManager.cancelAnimations()
             delegate?.gestureBegan(for: .pan)
         case .changed:
-            guard let initialTouchLocation = initialTouchLocation,
-                  let initialCameraState = initialCameraState,
-                  let panScrollingMode = delegate?.panScrollingMode else {
-                return
-            }
+            lastChangedDate = dateProvider.now
             cameraAnimationsManager.cancelAnimations()
-
-            // Reset the camera to its state when the gesture began
-            mapboxMap.setCamera(to: CameraOptions(cameraState: initialCameraState))
-
-            let clampedTouchLocation: CGPoint
-            switch panScrollingMode {
-            case .horizontal:
-                clampedTouchLocation = CGPoint(x: touchLocation.x, y: initialTouchLocation.y)
-            case .vertical:
-                clampedTouchLocation = CGPoint(x: initialTouchLocation.x, y: touchLocation.y)
-            case .horizontalAndVertical:
-                clampedTouchLocation = touchLocation
+            handleChange(withTouchLocation: touchLocation)
+        case .ended:
+            // decelerate
+            guard let lastChangedDate = lastChangedDate,
+                  // if it's been more than 2 frames at 60 Hz since the last change, don't drift
+                  dateProvider.now.timeIntervalSince(lastChangedDate) < 2.0 / 60.0,
```

Comment:
It's a bit confusing to me to describe this logic in terms of "frames at frequency" (especially since 60Hz doesn't seem to have any real significance). Would it be clearer to just store `2.0 / 60.0` as some timeout constant that is consumed as part of this conditional?

Reply:
I'm fine with that.